internal/fs/fs_test.go

```diff
@@ -772,34 +767,39 @@ func TestIsEmpty(t *testing.T) {
 	h := test.NewHelper(t)
 	defer h.Cleanup()
 
+	var inaccessibleDir string
+	cleanup := setupInaccessibleDir(t, func(dir string) error {
+		inaccessibleDir = filepath.Join(dir, "empty")
+		return os.Mkdir(inaccessibleDir, 0777)
+	})
+	defer cleanup()
+
 	h.TempDir("empty")
-	tests := map[string]string{
-		wd:                                                  "true",
-		"testdata":                                          "true",
-		filepath.Join(wd, "fs.go"):                          "err",
-		filepath.Join(wd, "this_file_does_not_exist.thing"): "false",
-		h.Path("empty"):                                     "false",
-	}
 
-	for f, want := range tests {
-		empty, err := IsNonEmptyDir(f)
-		if want == "err" {
-			if err == nil {
-				t.Fatalf("Wanted an error for %v, but it was nil", f)
-			}
-			if empty {
-				t.Fatalf("Wanted false with error for %v, but got true", f)
+	tests := []struct {
+		path  string
+		empty bool
+		err   bool
+	}{
+		{wd, true, false},
+		{"testdata", true, false},
+		{filepath.Join(wd, "fs.go"), false, true},
+		{filepath.Join(wd, "this_file_does_not_exist.thing"), false, false},
+		{h.Path("empty"), false, false},
+		{inaccessibleDir, false, false},
```

Comment:
`setupInaccessibleDir` doesn't work on Windows due to how permissions work there.
This test case should be skipped on Windows for now.